### PR TITLE
feat: add variant_id field to CourseRun model to support external LOBs

### PR DESCRIPTION
**Pull request by AfaqShuaib09**
[PROD-3733](https://2u-internal.atlassian.net/browse/PROD-3733)
--------------------

This PR adds the `variant_id` field  to CourseRun model to support External LOBs (ExecEd, Bootcamps).

### API response

image

### CODE REVIEW

File: course_discovery/apps/course_metadata/models.py [hunk 1]
```diff
@@ -2160,6 +2160,11 @@ class CourseRun(ManageHistoryMixin, DraftModelMixin, CachedMixin, TimeStampedMod
         help_text=_('This calculated field signifies if this course run is in the enterprise subscription catalog'),
     )
 
+    variant_id = models.UUIDField(
+        blank=False, null=True, editable=True,
+        help_text=_('The identifier for a product variant.')
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/models.py]: The help text should be updated to help differentiate between this and already present uuid field in CourseRun.

File: course_discovery/apps/course_metadata/toggles.py [hunk 2]
```diff
@@ -54,3 +54,14 @@
 IS_SUBDIRECTORY_SLUG_FORMAT_FOR_BOOTCAMP_ENABLED = WaffleSwitch(
     'course_metadata.is_subdirectory_slug_format_for_bootcamp_enabled', __name__
 )
+# .. toggle_name: course_metadata.is_course_run_variant_id_editable
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: Enable to make course run variant id editable (for external LOBs).
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/toggles.py]: nit: Please mention that this is only for CourseRun API.

**Review by DawoudSheraz — APPROVED** [review]
A few small comments, otherwise looks good. I have not tested it on local but the changes look good enough.

File: course_discovery/apps/course_metadata/toggles.py [hunk 3]
```diff
@@ -54,3 +54,14 @@
 IS_SUBDIRECTORY_SLUG_FORMAT_FOR_BOOTCAMP_ENABLED = WaffleSwitch(
     'course_metadata.is_subdirectory_slug_format_for_bootcamp_enabled', __name__
 )
+# .. toggle_name: course_metadata.is_course_run_variant_id_editable
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: Enable to make course run variant id editable via CourseRun API (for extetnal LOBs).
```
Comment (zawan-ila) [inline course_discovery/apps/course_metadata/toggles.py]: [spelling] external*

File: course_discovery/apps/course_metadata/toggles.py [hunk 4]
```diff
@@ -54,3 +54,14 @@
...
+# .. toggle_use_cases: open_edx
+# .. toggle_creation_date: 2023-09-05
```
Comment (zawan-ila) [inline course_discovery/apps/course_metadata/toggles.py]: Is this date correct?

File: course_discovery/apps/course_metadata/models.py [hunk 5]
```diff
@@ -2160,6 +2160,12 @@ class CourseRun(ManageHistoryMixin, DraftModelMixin, CachedMixin, TimeStampedMod
         help_text=_('This calculated field signifies if this course run is in the enterprise subscription catalog'),
     )
 
+    variant_id = models.UUIDField(
+        blank=False, null=True, editable=True,
```
Comment (zawan-ila) [inline course_discovery/apps/course_metadata/models.py]: [think] Have you tried saving a course run for an OC course from the admin? I believe the `blank=False` here would raise an error?

Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/models.py]: Nice catch, updating it.

File: course_discovery/apps/course_metadata/toggles.py (hunk 4 above)
Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/toggles.py]: let me update it, I added the wrong month before.

**Review by zawan-ila — APPROVED** [review]

**Pull request merged**